datacube/testutils/io.py

```diff
@@ -132,32 +137,168 @@ def native_geobox(ds: Dataset, measurements=None, basis: str | None = None):
     return geobox
 
 
-def native_load(ds, measurements=None, basis=None, **kw):
-    """Load single dataset in native resolution.
+def compute_native_load_geobox(
+    ds: Dataset,
+    band: str,
+    dst_geobox: GeoBox | None = None,
+    buffer: float | None = None,
+) -> GeoBox:
+    """Compute area of interest for an input dataset given final output geobox.
+
+    Take native projection and resolution from ``ds, band`` pair and compute
+    region in that projection that fully encloses footprint of the
+    ``dst_geobox`` with some padding. Construct GeoBox that encloses that
+    region fully with resolution/pixel alignment copied from supplied band.
+
+    :param ds: Sample dataset (only resolution and projection is used, not footprint)
+    :param band: Reference band to use
+    :param dst_geobox:
+                 (resolution of output GeoBox will match resolution of this band)
+    :param buffer: Buffer in units of CRS of ``ds`` (meters usually),
+                   default is 10 pixels worth
+    """
+    native: GeoBox = native_geobox(ds, basis=band)
+    if dst_geobox is None:
+        return native
+
+    if buffer is None:
+        buffer = 10 * max(map(abs, (native.resolution.y, native.resolution.x)))
+
+    assert native.crs is not None
+    return GeoBox.from_geopolygon(
+        dst_geobox.extent.to_crs(native.crs).buffer(buffer),
+        crs=native.crs,
+        resolution=native.resolution,
+        align=native.alignment,
+    )
 
-    :param ds: Dataset
-    :param measurements: List of band names to load
+
+def _split_by_grid(xx: xr.DataArray) -> list[xr.DataArray]:
+    """Split datasets by grid/crs"""
+
+    def extract(grid_id, ii):
+        yy = xx[ii]
+        crs = xx.grid2crs[grid_id]
+        yy.attrs.update(crs=crs)
+        yy.attrs.pop("grid2crs", None)
+        return yy
+
+    return [extract(grid_id, ii) for grid_id, ii in xx.groupby(xx.grid).groups.items()]
+
+
+def _native_load_1(
+    sources: xr.DataArray,
+    bands: tuple[str, ...],
```

Comment:
Can this type be `Sequence[str]` instead, so the caller doesn't have to convert the argument to a tuple?